cmd/tsdb/main.go

```diff
@@ -58,16 +65,185 @@ func main() {
+			o := append(overBcks[:largest], overBcks[largest+1:]...)
+			// Add this range to all blocks for deletion.
+			blocksDel = append(blocksDel, o...)
+		}
+
+		var paths []string
+		for _, b := range blocksDel {
+			_, folder := path.Split(b.Dir())
+			if _, err := ulid.Parse(folder); err != nil {
+				fmt.Printf("\nskipping invalid block dir: %v :%v \n\n", b.Dir(), err)
+				continue
+			}
+			paths = append(paths, b.Dir())
+		}
+		printBlocks(blocksDel, hformat)
+		if confirm() {
+			if err = dellAll(paths); err != nil {
+				exitWithError(errors.Wrap(err, "deleting overlapping blocks"))
+			}
+		}
+	}
+}
+
+func scanIndexes(scan tsdb.Scanner, hformat *bool) {
+	unrepairable, repaired, err := scan.Indexes()
+	if err != nil {
+		exitWithError(err)
+	}
+
+	if len(repaired) > 0 {
+		fmt.Println("Corrupted indexes that were repaired.")
+		for _, stats := range repaired {
+			fmt.Printf("path:%v stats:%+v  \n", stats.BlockDir, stats)
+		}
+	}
+
+	if len(unrepairable) > 0 {
+		for cause, bdirs := range unrepairable {
+			fmt.Println("Blocks with unrepairable indexes! \n", cause)
+			printFiles(bdirs, hformat)
+			if confirm() {
+				if err = dellAll(bdirs); err != nil {
+					exitWithError(errors.Wrap(err, "deleting blocks with invalid indexes"))
+				}
+			}
+		}
+	}
+}
+
+func scanTmbst(scan tsdb.Scanner, hformat *bool) {
+	invalid, err := scan.Tombstones()
+	if err != nil {
+		exitWithError(errors.Wrap(err, "scannings Tombstones"))
+	}
+
+	if len(invalid) > 0 {
+		fmt.Println("Tombstones include data to be deleted so removing these will cancel deleting these timeseries.")
+		for cause, files := range invalid {
+			for _, p := range files {
+				_, file := filepath.Split(p)
+				if file != "tombstone" {
+					exitWithError(fmt.Errorf("path doesn't contain a valid tombstone filename: %v", p))
+				}
+			}
+			fmt.Println("invalid tombstones:", cause)
+			printFiles(files, hformat)
+			if confirm() {
+				if err = dellAll(files); err != nil {
+					exitWithError(errors.Wrap(err, "deleting Tombstones"))
+				}
+			}
+		}
+	}
+}
+
+func scanTmps(scanPath string, hformat *bool) {
+	var files []string
+	filepath.Walk(scanPath, func(path string, f os.FileInfo, _ error) error {
+		if filepath.Ext(path) == ".tmp" {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if len(files) > 0 {
+		fmt.Println(`
+			These are usually caused by a crash or incomplete compaction and 
+			are safe to delete as long as no other application is currently using this database.`)
+		for _, p := range files {
+			if filepath.Ext(p) != ".tmp" {
+				exitWithError(fmt.Errorf("path doesn't contain a valid tmp extension: %v", p))
+			}
+		}
+		printFiles(files, hformat)
+		if confirm() {
+			if err := dellAll(files); err != nil {
+				exitWithError(errors.Wrap(err, "deleting temp files"))
+			}
+		}
+	}
+}
+
+func dellAll(paths []string) error {
+	for _, p := range paths {
+		if err := os.RemoveAll(p); err != nil {
+			return fmt.Errorf("error deleting: %v, %v", p, err)
```

Comment:
I though `errors.Wrap` is preferable, but not sure what convention you want to follow everywhere